lib/perl/Genome/Model/ClinSeq/Command/UpdateAnalysis.pm

```diff
@@ -820,12 +840,32 @@ sub get_trsn{
 
   my %trsns;
   my $trsn_ref;
-
-  foreach my $instrument_data (@instrument_data){
-    my $trsn = $instrument_data->target_region_set_name;
-    if ($trsn){
-      $trsns{$trsn}=1;
-      $trsn_ref = $trsn;
+  
+  #When using validation-as-exome and my-trsn options, use user-defined target region set name (exome or validation)
+  if ($self->validation_as_exome && $self->my_trsn) {
```

Comment:
To answer your questions:
I am also assuming that there should not be more than one feature list with the same trsn and content type. I think that would cause other problems in the pipeline and at this time, I'm not really sure I know how to fix it.

The feature list is only used in reference to the $trsn but is grabbed for each piece of instrument data for different purposes in the code.

Condencing the code:
Unfortunately, I don't think the condensation of the code can happen this way. The way it used to work is it iterated through the exome instrument data and just returned the last trsn (but also warned the user if there was more than 1 type of trsn). With my new implementation, I'm combining two different data types (validation and exome) but I only want it to assign the trsn of the data type of the user's choice. If I combine data types like this and do the old method, it will only use the last trsn read in which will not be consistent across my cohort of samples. I still want it to assign 1 to %trsns for both types of instrument data for counting purposes but only the specified trsn to the $trsn_ref that will be returned at the end of the subroutine. I think your method will still assign the trsn of the last piece of instrument data to $trsn_ref. That is, if I understand your code.